File: app/src/main/java/com/example/cheng/myapplication/Point3View.java
package com.example.cheng.myapplication;

import android.animation.ValueAnimator;
import android.content.Context;
import android.graphics.Color;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.util.Log;
import android.view.animation.Animation;
import android.view.animation.LinearInterpolator;
import android.widget.TextView;

/**
 * Created by biao.cheng on 2017/10/24.
 */

public    class Point3View extends TextView {
    private ValueAnimator animator;

    public Point3View(Context context) {
        super(context);
        init();
    }
    public Point3View(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);init();
    }

    public Point3View(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);init();
    }

    private void init() {
        this.setText("...");
        this.setTextColor(Color.WHITE);
        this.setTextSize(20);
    }

    private int currentValue;

    public void showPoint(){
         animator = ValueAnimator.ofInt(0,3);
        animator.setInterpolator(new LinearInterpolator());
        animator.setDuration(2500);
        animator.setRepeatCount(100);
        animator.start();

        animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator valueAnimator) {
                int current = (int) valueAnimator.getAnimatedValue();
                if(current==currentValue){
                    return;
                }else {
                    currentValue=current;
                }
                Log.i("testView",current+"====");
                switch (current){
                    case 0:
                        setText(".");
                        break;
                    case 1:
                        setText("..");
                        break;
                    case 2:
                        setText("...");
                        break;
                }
            }
        });
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        showPoint();
    }

    @Override
    protected void onDetachedFromWindow() {
        super.onDetachedFromWindow();
        if(animator!=null){
            animator.setRepeatCount(0);
            animator.cancel();
            animator.cancel();
        }
    }
}

File: app/src/main/java/com/example/cheng/myapplication/Point3View.kt
package com.example.cheng.myapplication

import android.animation.ValueAnimator
import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.util.Log
import android.view.animation.Animation
import android.view.animation.LinearInterpolator
import android.widget.TextView

/**
 * Created by biao.cheng on 2017/10/24.
 */

class Point3View : TextView {
    private var animator: ValueAnimator? = null

    private var currentValue: Int = 0

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        this.text = "..."
        this.setTextColor(Color.WHITE)
        this.textSize = 20f
        main()
    }

    fun main(): Unit {

//        Log.i("String",it)

        "Strubf".run {
            Log.i("setTextColor", this.toString())
        }.let {
            Log.i("setTextColor", it.toString())
        }

    }

    fun showPoint() {
        animator = ValueAnimator.ofInt(0, 3)
        animator!!.interpolator = LinearInterpolator()
        animator!!.duration = 2500
        animator!!.repeatCount = 100
        animator!!.start()

        animator!!.addUpdateListener(ValueAnimator.AnimatorUpdateListener { valueAnimator ->
            val current = valueAnimator.animatedValue as Int
            if (current == currentValue) {
                return@AnimatorUpdateListener
            } else {
                currentValue = current
            }
            Log.i("testView", current.toString() + "====")
            when (current) {
                0 -> text = "."
                1 -> text = ".."
                2 -> text = "..."
            }
        })
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        showPoint()
    }

    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
        if (animator != null) {
            animator!!.repeatCount = 0
            animator!!.cancel()
            animator!!.cancel()
        }
    }
}
